pkg/scalers/mongo_scaler.go

```diff
@@ -189,8 +190,7 @@ func parseMongoDBMetadata(config *ScalerConfig) (*mongoDBMetadata, string, error
 	} else {
 		// Build connection str
 		addr := net.JoinHostPort(meta.host, meta.port)
-		auth := fmt.Sprintf("%s:%s", meta.username, meta.password)
-		connStr = "mongodb://" + auth + "@" + addr + "/" + meta.dbName
+		connStr = fmt.Sprintf("mongodb://%s:%s@%s/%s", url.QueryEscape(meta.username), url.QueryEscape(meta.password), addr, meta.dbName)
```

Comment:
Can you add this to the changelog please?

Reply:
Does it make sense? It's and small fix which is transparent for users

Reply:
Yes because it's a (security) improvement which has impact on scaler end-users as their configuration is now used differently (and thus might lead to changes)

Reply:
done